docs/source/tutorial/understanding_evaluation.rst

```diff
@@ -180,6 +180,75 @@ may be additional known triples :math:`(h, r, t')` for :math:`t \neq t'`. If the
 filtered evaluation setting ignores for a given triple :math:`(h, r, t)` the scores of all other *known* true triples
 :math:`(h, r, t')`.
 
+Below, we present the philosophy from [bordes2013]_ and how it is implemented in PyKEEN:
+
+HPO Scenario
+************
+During training/optimization with :func:`pykeen.hpo.hpo_pipeline`, the set of known positive triples comprises the
+training and validation sets. After optimization is finished and the final evaluation is done, the set of known
+positive triples comprises the training, validation, and testing set. PyKEEN explicitly does not use test triples
+for filtering during HPO avoid any test leakage.
+
+Early Stopper Scenario
+**********************
+When early stopping is used during training, it periodically uses the validation set for calculating the loss
+and evaluation metrics. During this evaluation, the set of known positive triples comprises the training and
+validation sets. When final evaluation is done with the testing set, the set of known positive triples comprises the
```

Comment:
Is this the case? Where does the early stopper get the test triples from?

Reply:
Currently it does use the validation set, see:

https://github.com/pykeen/pykeen/blob/e77415225784ef11da9d2eabd6587c99d70ffdb1/src/pykeen/pipeline/api.py#L946-L954

If this is methodologically wrong, we can update it

Reply:
This part shows it is using the *validation* triples, which is correct :+1: 

I was wondering about the part
```
When final evaluation is done with the testing set, the set of known positive triples comprises the
```
in the explanation about the early stopper.